src/services/fs/backend.rs

```diff
@@ -83,16 +101,16 @@ impl Builder {
     pub fn build(&mut self) -> Result<impl Accessor> {
         debug!("backend build started: {:?}", &self);
 
-        let root = normalize_root(&self.root.take().unwrap_or_default());
-        let atomic_write_dir = self.atomic_write_dir.as_deref().map(normalize_root);
+        let root = self.root.take().unwrap_or_default();
```

Comment:
root must be [`canonicalize`](https://doc.rust-lang.org/std/path/struct.Path.html#method.canonicalize), we will not accept `../` or `/../../path` as root.

Reply:
The problem with `canonicalize` is that the path must already exist. So the code for creating the directory won't work.

Reply:
Ok, we can create dir first and than [canonicalize](https://doc.rust-lang.org/std/path/struct.Path.html#method.canonicalize) them.

Reply:
Maybe you can use https://crates.io/crates/path-clean ?

Reply:
Seems cool, but I prefer to include the implementation directly instead of depend on it. We can do this in another PR.